### PR TITLE
Ruby 3 compatibility

### LINKED ISSUE #190: Ruby 3.0 support for broadcast with kwargs
When broadcasting with kwargs, an `ArgumentError` is raised in Ruby 3.0:

```ruby
broadcast(:user_changed, initial_id: user_id_before_last_save, final_id: user_id)
```

```ruby
ArgumentError: wrong number of arguments (given 1, expected 0; required keywords: initial_id, final_id)
```

This worked fine on Ruby 2.6, but because `Publisher#broadcast` doesn't handle kwargs, it breaks in 3.0.

### PR DESCRIPTION
**Pull request by merrington**
Fixes #190.

This picks up on the work from #194 with a few minor additions.
- Addressed the feedback provided by @oskarpearson in #194 
- Added/fixed some specs to highlight differences between ruby 2.7 and ruby 3
- Bumped major version, minor readme changes

~For now I've added `jruby` support as "experimental" into the CI pipeline. It does appear to work, except for https://github.com/jruby/jruby/issues/7112 which is causing one spec to fail with jruby 9.4. Presumably once this patch is released/available, the specs should pass for jruby as well.~ This has been fixed

---

I also have related changes to some of the wisper-* projects (namely wisper-sidekiq, wisper-testing and wisper-rspec) to support ruby 3. Happy to open PR's for those as well if this looks good.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          - ruby: jruby
+            experimental: true
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - run: bundle exec rspec spec
```
Comment (thirunjuguna) [inline .github/workflows/build.yml:25]:
```suggestion
      - run: bundle exec rspec spec
 
```

**Comment by merrington** [comment]
@krisleech - sorry to bother you, but wondering if you'll have a chance to look at this?

**Comment by firstpromoter** [comment]
nice job @merrington

**Comment by joevandyk** [comment]
would love to see this merged in!

File: lib/wisper/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Wisper
-  VERSION = "2.0.1"
+  VERSION = "3.0.0"
```
Comment (nashby) [inline lib/wisper/version.rb:2]: could you please revert this change? it's better to leave it to maintainers to decide when gem's version has to be updated

**Comment by krisleech** [comment]
Looks good, for anyone upgrading from version 2.x to 3.x do they need to do anything? (assuming they aren't using a Ruby smaller than 2.7)

File: lib/wisper/version.rb (hunk 2 above)
Comment (krisleech) [reply lib/wisper/version.rb:2]: Agreed, I can cut a release in a separate commit, just make sure to mark it as backwards incompatible in the CHANGELOG. Thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 ## HEAD (unreleased)
 
+## 3.0 (20th Jan 2023)
+Authors: doits, jstoks, merringtion
+
+* Adds: Support for Ruby 3.0 keyword arguments
+* Removes: Support for Ruby 2.6 and lower
+
```
Comment (krisleech) [inline CHANGELOG.md:8]: Can you remove the 3.0 heading, thanks.

Is the addition of Ruby 3.0 keyword arguments a breaking change, do those using Ruby 2.x need to make any changes?

**Comment by krisleech** [comment]
I'm also wondering if any changes need to be made to wisper-rspec, but I guess they can be done independently.

**Comment by merrington** [comment]
Thanks for the review - I will address this over the next day or two.

> I'm also wondering if any changes need to be made to wisper-rspec, but I guess they can be done independently.

I already have a [branch of wisper-rspec](https://github.com/merrington/wisper-rspec/tree/ruby-3) for ruby 3, I just wanted to make sure there would be no significant changes here before opening a PR for that. (I also have branches for wisper-testing and wisper-sidekiq). I'll get those opened in the next couple of days as well.

**Comment by krisleech** [comment]
I guess we can close https://github.com/krisleech/wisper/pull/192 too

**Comment by patrickemuller** [comment]
Hey everyone, how can I contribute to getting this merged as soon as possible?
@krisleech let me know how I can contribute.

**Comment by jrinaudo** [comment]
I can help too, I would love to have this merged ASAP. @merrington let me know if I can do anything to help. Thanks!

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        experimental: [false]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - run: bundle exec rspec spec
+
```
[the remark this answers is not in the document]
Comment (krisleech) [reply .github/workflows/build.yml:23]: I merged https://github.com/krisleech/wisper/pull/201 - hopefully no conflict with this.

**Comment by krisleech** [comment]
Couple of conflicts need resolving and the requested change to the CHANGELOG, then can be merged.

**Comment by krisleech** [comment]
I've fixed the conflicts on https://github.com/krisleech/wisper/pull/205
If CI passes then I will prepare an `3.0-rc1` release.
